service/matching/db.go

```diff
@@ -188,11 +204,24 @@ func (db *taskQueueDB) UpdateState(
 // CreateTasks creates a batch of given tasks for this task queue
 func (db *taskQueueDB) CreateTasks(
 	ctx context.Context,
-	tasks []*persistencespb.AllocatedTaskInfo,
+	taskIDs []int64,
+	reqs []*writeTaskRequest,
 ) (*persistence.CreateTasksResponse, error) {
 	db.Lock()
 	defer db.Unlock()
-	return db.store.CreateTasks(
+
+	maxReadLevel := int64(0)
+	var tasks []*persistencespb.AllocatedTaskInfo
+	for i, req := range reqs {
+		tasks = append(tasks, &persistencespb.AllocatedTaskInfo{
+			TaskId: taskIDs[i],
+			Data:   req.taskInfo,
+		})
+		maxReadLevel = taskIDs[i]
+	}
+
+	db.maxReadLevel = maxReadLevel
```

Comment:
in old implementation update is conditional on `maxReadLevel > 0` to be safe we need to keep that check or something similar in here.
Also, we're changing order of db.maxReadLevel assignment and bringing it to before CreateTask. Not sure it's desired.

Reply:
I had thought of this during it's implementation and would like to raise the question:
Is it possible, that we add tasks, and the check `maxReadLevel > 0` fails? 

After spending some time thinking about this, I found out that the answer to that is when a request would have 0 tasks for addition. In that case, `maxReadLevel` would remain 0 since we do not iterate over any tasks `but` still send this request all the way to `db`. Do we want this behaviour?

Additionally, I don't think the check `maxReadLevel > 0` should have existed in the first place (if we do not allow for empty tasks) since we are guaranteeing that each newer task would have taskID which is increasing in nature, and `maxReadLevel` would be > 0 effectively.

tldr; 

1. Do we still want to have functionality to send in requests to `db` when we have 0 tasks in the array? If yes, we have to add in a check to see how `maxReadLevel` gets updated inside of db.

Reply:
right, I think it's fine two skip the write if the list is empty. but that needs to be added at the beginning of CreateTasks and return from the method, right?

Reply:
I agree; shall add it thanks